stepactions/get-unreleased-bundle/0.1/get-unreleased-bundle.yaml

```diff
@@ -0,0 +1,103 @@
+    If the bundle is unreleased, the StepAction returns it.
+  image: quay.io/konflux-ci/konflux-test:v1.4.15@sha256:23a7e8c745a037da283f18f5e5108598efeaeb5263d599411b6c4a25c42c6f94
+  params:
+    - name: fbcFragment
+      type: string
+      description: A FBC fragment image.
+    - name: packageName
+      type: string
+      description: Optional. An OLM package name present in the fragment or leave it empty so the step will determine the default package name as described above.
+      default: ""
+    - name: channelName
+      type: string
+      description: Optional. An OLM channel name or leave it empty so the step will determine the default channel name as described above.
+      default: ""
+  results:
+    - name: unreleasedBundle
+      description: The name of the bundle that is not in registry.redhat.io/redhat/redhat-operator-index.
+  env:
+    - name: FBC_FRAGMENT
+      value: $(params.fbcFragment)
+    - name: PACKAGE_NAME
+      value: $(params.packageName)
+    - name: CHANNEL_NAME
+      value: $(params.channelName)
+  script: |
+    #!/usr/bin/env bash
+    set -euo pipefail
+    . /utils.sh
+
+    if [ -z "$FBC_FRAGMENT" ]; then
+      echo "Error: FBC_FRAGMENT parameter is required." >&2
+      exit 1
+    fi
+
+    # Render the FBC fragment
+    if ! RENDER_OUT_FBC=$(render_opm -t "$FBC_FRAGMENT"); then
+      echo "Failed to render the FBC fragment" >&2
+      exit 1
+    fi
+
+    # Determine packageName if not provided
+    if [ -z "$PACKAGE_NAME" ]; then
+      PACKAGE_NAME=$(get_package_from_catalog "$RENDER_OUT_FBC")
+      if [ -z "$PACKAGE_NAME" ]; then
+        echo "Failed to determine a default package name" >&2
+        exit 1
+      fi
+      echo "Package name is '$PACKAGE_NAME'"
+    fi
+
+    # Determine channelName if not provided
+    if [ -z "$CHANNEL_NAME" ]; then
+      CHANNEL_NAME=$(get_channel_from_catalog "$RENDER_OUT_FBC" "$PACKAGE_NAME")
+      if [ -z "$CHANNEL_NAME" ]; then
+        echo "Failed to determine a default channel name for package '$PACKAGE_NAME'" >&2
+        exit 1
+      fi
+      echo "Default channel name for package '$PACKAGE_NAME' is '$CHANNEL_NAME'"
+    fi
+
+    # Get highest bundle version from a given package and channel
+    if ! highest_bundle=$(get_highest_bundle_version "$RENDER_OUT_FBC" "$PACKAGE_NAME" "$CHANNEL_NAME"); then
+      echo "Failed to determine a highest bundle version for package '$PACKAGE_NAME' and channel '$CHANNEL_NAME'" >&2
+      exit 1
+    fi
+    echo "Highest bundle version for package '$PACKAGE_NAME' and channel '$CHANNEL_NAME' is '$highest_bundle'"
+
+    # Find the unreleased bundle(s)
+    if ! unreleased_bundles=$(get_unreleased_bundles -i "$FBC_FRAGMENT"); then
+      echo "Could not get unreleased bundle images from the fragment. Make sure you have ImagePullCredentials for registry.redhat.io" >&2
+      exit 0
```

Comment:
not sure if this is an error, so wondering if `exit 1` should be here instead?

Reply:
good catch, it definitely should be `exit 1`